packages/react-dom/src/__tests__/ReactDOMServerSelectiveHydration-test.internal.js

```diff
@@ -1770,4 +1785,71 @@ describe('ReactDOMServerSelectiveHydration', () => {
 
     document.body.removeChild(container);
   });
+
+  describe('it can force hydration in response to', () => {
+    let spanRef;
+    let initialSpan;
+    let root;
+    let App;
+    beforeEach(() => {
+      function Child({text}) {
+        Scheduler.unstable_yieldValue(`Child ${text}`);
+        return <span ref={ref => (spanRef = ref)}>{text}</span>;
+      }
+      App = function({text}) {
+        Scheduler.unstable_yieldValue(`App ${text}`);
+        return (
+          <div>
+            <Suspense fallback={null}>
+              <Child text={text} />
+            </Suspense>
+          </div>
+        );
+      };
```

Comment:
Nit: We try to avoid beforeEach except for importing modules. If you want to reuse these component definitions, you don't need `beforeEach` —  you can put them in module scope.

However, personally, I prefer to copy paste them between tests because it makes debugging the tests later much easier.

It's more important that tests are easy to read, not easy to write.